### PR TITLE
docs: Add test skeleton 'test_setting_concurrent_rebuild_limit'

### PR DESCRIPTION
**Pull request by shuo-wu**
Longhorn/longhorn#2460

### CODE REVIEW

File: manager/integration/tests/test_settings.py [hunk 1]
```diff
@@ -658,3 +658,49 @@ def test_setting_backing_image_auto_cleanup():  # NOQA
        and all entries will be removed from the map later.
     9. Delete the backing image.
     """
+
+
+@pytest.mark.skip(reason="TODO")
+@pytest.mark.backing_image  # NOQA
```
Comment (khushboo-rancher) [inline manager/integration/tests/test_settings.py:664]: Why is this marked as backing_image?

File: manager/integration/tests/test_settings.py [hunk 2]
```diff
@@ -658,3 +658,49 @@ def test_setting_backing_image_auto_cleanup():  # NOQA
...
+    5. Verify the new replica of volume 2 won't be started until volume 1
+       rebuilding complete.
+       And the new replica of volume 2 will be started immediately once
+       the 1st rebuilding is done.
+    6. Wait for rebuilding complete then repeat step 4.
+    7. Set `ConcurrentReplicaRebuildPerNodeLimit` to 0 or 2 while the volume 1
+       rebuilding is still in progress.
+       Then the new replica of volume 2 will be started immediately before
+       the 1st rebuilding is done.
+
```
Comment (khushboo-rancher) [inline manager/integration/tests/test_settings.py:687]: Should we add a case where the replica rebuilding is failed for the volume 1 while the setting `ConcurrentReplicaRebuildPerNodeLimit` to 1?

In this case, I'm assuming the replica rebuilding for volume 2 should start as failure case should not stop other rebuilding to progress.

File: manager/integration/tests/test_settings.py (hunk 1 above)
Comment (shuo-wu) [reply manager/integration/tests/test_settings.py:664]: Sorry, typo here...

File: manager/integration/tests/test_settings.py (hunk 2 above)
Comment (shuo-wu) [reply manager/integration/tests/test_settings.py:687]: Yes, good suggestion! Let me add this case to the skeleton.

**Review by innobead — APPROVED** [review]
LGTM

**Review by khushboo-rancher — APPROVED** [review]
Thanks @shuo-wu for adding the failed replica scenario.

**Pull request merged**